app/main.py

```diff
@@ -1,2 +1,90 @@
+from dataclasses import dataclass
+from typing import Hashable, Iterator
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: any
+    hash_value: int
+
+
 class Dictionary:
-    pass
+    INITIAL_CAPACITY = 8
+    THRESHOLD = 2 / 3
+    CAPACITY_MULTIPLIER = 2
+
+    def __init__(self, initial_capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = initial_capacity
+        self.size = 0
+        self.buckets = [[] for _ in range(self.capacity)]
+
+    def _get_index(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def _resize(self) -> None:
+        new_capacity = self.capacity * self.CAPACITY_MULTIPLIER
+        new_buckets = [[] for _ in range(new_capacity)]
+
+        for bucket in self.buckets:
+            for node in bucket:
+                new_index = node.hash_value % new_capacity
+                new_buckets[new_index].append(node)
+
+        self.capacity = new_capacity
+        self.buckets = new_buckets
+
+    def __setitem__(self, key: Hashable, value: any) -> None:
+        if isinstance(key, (list, dict, set)):
```

Comment:
The check for whether the key is an instance of list, dict, or set is redundant because the typing.Hashable already ensures that the key is hashable. Consider removing this check.